pkg/noun/jets/c/aor.c

```diff
@@ -0,0 +1,134 @@
+/// @file
+
+#include "jets/q.h"
+#include "jets/w.h"
+
+#include "noun.h"
+#include "portable.h"
+
+#if defined(c3_endian) && defined(c3_endian_little) && (c3_endian == c3_endian_little)
+
+  u3_noun
+  u3qc_aor(u3_noun a,
+           u3_noun b)
+  {
+    while ( 1 ) {
+      if ( c3y == u3r_sing(a, b) ) return c3y;
+      if ( c3n == u3ud(a) ) {
+        if ( c3y == u3ud(b) ) return c3n;
+        if ( c3y == u3r_sing(u3h(a), u3h(b)) ) {
+          a = u3t(a);
+          b = u3t(b);
+        }
+        else {
+          a = u3h(a);
+          b = u3h(b);
+        }
+      }
+      else {
+        if ( c3n == u3ud(b) ) return c3y;
+        {
+          c3_w len_a_y, len_b_y;
+          c3_y *a_bytes, *b_bytes;
+          c3_y a_y, b_y;
+          if ( c3y == u3a_is_cat(a) ) {
+            len_a_y = 4;
+            a_bytes = (c3_y*)&a;
+          }
+          else {
+            u3a_atom* a_u = u3a_to_ptr(a);
+            len_a_y = (a_u->len_w)*4;
```

Comment:
This is wrong (as are your other length calculations in this function). The byte-length is not `4 * word-length`, it needs to precisely measure the most-significant word. You you use `u3r_met(3, a)` or calculate yourself.

Reply:
my bad, fixed it; in my defense this didn't affect the result since in the case of different byte size but same word size the function would've returned on comparison of a non-zero byte with one of the leading zero bytes instead of length comparison after the loop